python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -369,30 +370,35 @@ def makeMinimalSchema(filterNameList, *, addCentroid=False,
 
             - If 2 or 3: add fields "coord_raErr" and "coord_decErr".
             - If 3: also add field "coord_radecErr".
+
+            Ignored if ``fullPositionCovariance=True``.
         addProperMotion : `bool`
             If True add fields "epoch", "pm_ra", "pm_dec" and "pm_flag".
         properMotionErrDim : `int`
-            Number of proper motion error fields; must be one of 0, 2, 3;
-            ignored if addProperMotion false:
-            - If 2 or 3: add fields "pm_raErr" and "pm_decErr".
-            - If 3: also add field "pm_radecErr".
+            Deprecated (see DM-?????): Use ``fullPositionCovariance`` when proper motions
+            are provided, as they will be correlated with the other position
+            paramters.
         addParallax : `bool`
             If True add fields "epoch", "parallax", "parallaxErr"
             and "parallax_flag".
+        fullPositionCovariance : `bool`
+            Include a full coordinate+PM+parallax covariance matrix. Requires
+            that `addProperMotion` and `addParallax` are set.
+            Ignores the `coordErrDim` and  `properMotionErrDim` parameters.
 
         Returns
         -------
         schema : `lsst.afw.table.Schema`
             Schema for reference catalog, an
             `lsst.afw.table.SimpleCatalog`.
-
-        Notes
-        -----
-        Reference catalogs support additional covariances, such as
-        covariance between RA and proper motion in declination,
-        that are not supported by this method, but can be added after
-        calling this method.
         """
+        if properMotionErrDim is not None:
+            warnings.warn("properMotionErrDim is deprecated and has no effect; "
+                          "use `fullPositionCovariance` to include all position errors in the schema. "
+                          "It will be removed after release 25.0.", category=FutureWarning)
```

Comment:
Because there's no use for the current system: these hooks were in place to allow someone to put those error fields in the refcat, but we didn't actually have any code that used the `=3` case. Trying to maintain the above functionality (which uses different names for the fields) along with the new full covariance seemed like excessive work for no gain. The only relevant astrometric catalog is Gaia, and until this ticket we couldn't actually load the relevant error fields for Gaia; I'm positive that removing that old functionality breaks nobody's code.